hazelcast-client/src/test/java/com/hazelcast/client/test/TestClientRegistry.java

```diff
@@ -159,12 +154,22 @@ protected ClientConnection createSocketConnection(Address address) throws IOExce
             }
         }
 
+        private LockPair getLockPair(Address address) {
+            return ConcurrencyUtil.getOrPutIfAbsent(addressBlockMap, address,
+                                new ConstructorFunction<Address, LockPair>() {
+                                    @Override
+                                    public LockPair createNew(Address arg) {
+                                        return new LockPair(new ReentrantReadWriteLock(), new ReentrantReadWriteLock());
+                                    }
+                                });
+        }
+
         /**
          * Blocks incoming messages to client from given address
          */
         void blockFrom(Address address) {
             LOGGER.info("Blocked messages from " + address);
-            TwoWayBlockableExecutor.LockPair lockPair = addressBlockMap.get(address);
+            TwoWayBlockableExecutor.LockPair lockPair = getLockPair(address);
```

Comment:
Updated in the new PR for client tests : https://github.com/hazelcast/hazelcast/pull/10030